airbyte-webapp/src/views/Connection/CatalogDiffModal/utils.tsx

```diff
@@ -0,0 +1,27 @@
+import { FieldTransform, StreamTransform } from "core/request/AirbyteClient";
+
+import { SortedDiff } from "./types";
+
+export const getSortedDiff = <T extends StreamTransform | FieldTransform>(diffArray?: T[]): SortedDiff<T> => {
+  const sortedDiff: SortedDiff<T> = { newItems: [], removedItems: [], changedItems: [] };
+  if (!diffArray) {
+    return sortedDiff;
+  }
+
+  diffArray.reduce((sortedDiff, transform) => {
```

Comment:
This does not need a reduce. The forEach should be ok since it's already mutating the sortedDiff object.